client/question/services/code_evaluators/PythonPrereqCheckService.js

```diff
@@ -63,22 +87,31 @@ tie.factory('PythonPrereqCheckService', [
      *    WRONG_LANGUAGE_ERRORS.
      */
     var detectAndGetWrongLanguageType = function(code) {
-      var codeLines = getNonStringLines(code);
+      var obscuredCode = getObscuredCode(code);
+      var rawCodeArray = code.split('\n');
 
       for (var i = 0; i < WRONG_LANGUAGE_ERRORS.python.length; i++) {
         var error = WRONG_LANGUAGE_ERRORS.python[i];
         var regexp = new RegExp(error.regExString);
 
-        if (error.allowMultiline) {
-          if (code.search(regexp) !== -1) {
-            return error.errorName;
-          }
-        } else {
-          for (var j = 0; j < codeLines.length; j++) {
-            if (codeLines[j].search(regexp) !== -1) {
-              return error.errorName;
-            }
+        // Lookup character location of error
+        var firstErrorColumnNumber = obscuredCode.search(regexp);
+        if (firstErrorColumnNumber !== -1) {
+          var firstErrorLineNumber = 0;
+
+          // After regex matches an error, loop through codelines
+          for (var l = 0; firstErrorColumnNumber >= 0; l++) {
+            // Subtract l.length from firstErrorCharIndex.
+            // At/below 0: index === linenumber. Subtract +1 for newline char.
+            firstErrorColumnNumber -= (rawCodeArray[l].length + 1);
+            firstErrorLineNumber++;
           }
+
+          firstErrorColumnNumber += rawCodeArray[
```

Comment:
At this point it's reasonable to say firstErrorColumnNumber = firstErrorCharIndex + ...

Reply:
The loop ends when firstErrorColumnNumber >= 0, so without this line, The Column number ends up a negative string index.

Reply:
Makes sense. Done.